[[Bug 17794]] - add syntax entry including "else if" and "end if" in "if" control structure

MartinKoob opened the pull request:
I have added a syntax entry in the "if" control structure documentation and have modified the Summary and Description entries to include those terms as well.   Also added descriptions to the Parameter entries.

This may need further changes.  

- In the Description it mentions 4 forms of the if control structure showing various places that a line break is needed but in the dictionary they are displayed on one line.  The lines should be formatted with the line breaks included.  What is the best way to do that?

- The 4 forms don't include else if keywords.  Should a form be added to show those?

MartinKoob committed:
Bugfix-17794 update if..lcdoc

Make changes suggested in Bug #17794

MartinKoob committed:
Update if.lcdoc

Update the Summary, Description and Condition Parameter.  Trying to ensure that the links in the <statements|statemenList> are in the correct format.

MartinKoob committed:
Update if.lcdoc

Further modifications to summary ensuring if is properly marked up with <if> to ensure it will be shown in bold in dictionary,  add explanation of requirement for 'end if'  keyword in the summary.  Added reference to 'else if' and a link to that.  There is no "elseIf.lcdoc" file for those to reference however.  I have submitted a bug report #19786 - Dictionary does not contain 'else if' keyword entry.

MartinKoob committed:
Add release note

MartinKoob committed:
update if.lcdoc

Try shortening lines to see if that helps in passing the travis build.

MartinKoob commented:
The Travis CI build failed.  I thought it may be that some lines were longer than 72 characters.  I shortened some lines that appeared too long.

montegoulding commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -27,29 +43,40 @@ end if
+<statement list|statementList> following that <condition> is
+skipped.)
+
+statement|statementList:
```
@MartinKoob  this should be just `statementList:` Same below.

MartinKoob committed:
fix errors in if.lcdoc

fixed the param names by removing the portion before the '|'

MartinKoob committed:
update if.lcdoc

added  second 'else if' to link after |  <else if|else if> so the text will display as there is no 'else if' .lcdoc file to link to.  see Bug #19786

montegoulding commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -4,9 +4,25 @@ Type: control structure
 Summary:
-<execute|Executes> a list of <statement|statements> if a condition is
-true. 
+A <conditional statement> following an <if> or an optional series of
```
@MartinKoob wondering if this could be reworded so we don't have `conditional statement` so many times in a sentence. Also I think `<conditional expression|condition>` would be more accurate here. Probably the optional `else if` could be moved to a different sentence.

montegoulding commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -27,29 +43,40 @@ end if
 
     if condition then
-       statementList 
+        statementList 
```
This is out of line with indenting below now

MartinKoob commented:
Hi Monte 

OK I will make the changes. I may have time tomorrow.

Thanks for reviewing.

MartinKoob committed:
Made changes to Summary in response to Monte

I rewrote the summary section. I looked over the descriptions in the old HyperCard Script Language guide and Hypertalk 2.2 the book for some inspiration.  Hope this is clearer.

MartinKoob committed:
Rewrite of Description paragraphs - else & else if

I added to the Description updating the paragraph for the else keyword and adding a paragraph for the else if keyword.

MartinKoob committed:
Changed indenting as recommended by Monte

MartinKoob commented:
I have also created a pull request #5888 to create an else-if.lcdoc file to go along with this PR.

livecodeali commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -4,9 +4,30 @@ Type: control structure
+Evaluates a <conditional expression|condition> to determine if it is true
+and if so <execute|executes> a subsequent <statment|statement> or <statementlist|statement list>.
+
+If the <conditional expression|condition> is false and the <if> <control structure> contains
```
this should be <condition|conditional expression> - so that it refers to the condition parameter but is rendered as 'conditional expression'.

livecodeali commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -4,9 +4,30 @@ Type: control structure
-<execute|Executes> a list of <statement|statements> if a condition is
-true. 
+Evaluates a <conditional expression|condition> to determine if it is true
+and if so <execute|executes> a subsequent <statment|statement> or <statementlist|statement list>.
```
this should just be `<statement>` as `<A|A>` is equivalent to `<A>` (note the LHS here has a typo, missing an 'e')

livecodeali commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -4,9 +4,30 @@ Type: control structure
+and if so <execute|executes> a subsequent <statment|statement> or <statementlist|statement list>.
+
+If the <conditional expression|condition> is false and the <if> <control structure> contains
+an <else> keyword the <statement|statement> or <statementlist|statement list> following the <else>
```
`<statement>`

livecodeali commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -4,9 +4,30 @@ Type: control structure
+an <else> keyword the <statement|statement> or <statementlist|statement list> following the <else>
+<keyword> is executed.
+
+The  <if> <control structure> may contain 1 or more <else if|else if> <keyword|keywords> which have a
```
`<else if>`

livecodeali replied:
There's an extra space here between `The` and `<if>`. Also, could you put `one` instead of `1` please?

livecodeali commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -27,22 +48,33 @@ end if
+Any <expression>. (If the <condition> <evaluate|evaluates> to true, 
+the <statement|statements> execute for that <condition>.
+If the <condition> <evaluate|evaluates> to false, the <statement> or 
+<statement list|statementList> following that <condition> is
```
`<statementList|statement list>`

livecodeali commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -27,22 +48,33 @@ end if
-Use the <if> <control structure> to <execute> a <statementList|statement> 
-(or list of <statementList|statements>) only under certain circumstances. 
+Use the <if> <control structure> to <execute> a <statement>
+(or <list of statements|statementList>) only under certain circumstances. 
```
`<statementList|list of statements>`

livecodeali commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -27,22 +48,33 @@ end if
+(or <list of statements|statementList>) only under certain circumstances. 
 
-The <if> <control structure> always begins with the <word> if. There are four forms of the <if> <control structure>:
+The <if> <control structure> always begins with the <word> if. There
```
I don't think this `<word>` should be a reference, as it is just the word 'word' rather than the keyword `word`

livecodeali commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -72,9 +104,16 @@ If the <condition> <evaluate|evaluates> to true, the statement or
+contains an <else> <keyword>, the <elseStatement> or <elseStatementList>
+which follows it is <execute|executed>.
+
+If the <if> <control structure> contains 1 or more <else if>
```
Could you write `one` instead of `1` here please?

livecodeali commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -72,9 +104,16 @@ If the <condition> <evaluate|evaluates> to true, the statement or
+
+If the <if> <control structure> contains 1 or more <else if>
+<keyword|keywords> which have a <conditional expression|condition>
+and one of those <conditional expression|conditions> evaluates to be
```
`<condition|conditions>` or `<condition|conditional expressions>`

livecodeali commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -72,9 +104,16 @@ If the <condition> <evaluate|evaluates> to true, the statement or
+which follows it is <execute|executed>.
+
+If the <if> <control structure> contains 1 or more <else if>
+<keyword|keywords> which have a <conditional expression|condition>
```
`<condition|conditional expression>`

livecodeali commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -72,9 +104,16 @@ If the <condition> <evaluate|evaluates> to true, the statement or
+If the <if> <control structure> contains 1 or more <else if>
+<keyword|keywords> which have a <conditional expression|condition>
+and one of those <conditional expression|conditions> evaluates to be
+true the respective <elseIfstatment|else if statement> or
```
missed an 'e' here in `elseIfStatement`

livecodeali commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -72,9 +104,16 @@ If the <condition> <evaluate|evaluates> to true, the statement or
+and one of those <conditional expression|conditions> evaluates to be
+true the respective <elseIfstatment|else if statement> or
+<elseIfstatementlist|else if statement list> following that
+<else if|else if> <keyword> is executed.
```
this can just be `<else if>`

MartinKoob committed:
Update if.lcdoc to implement requested changes.

Thanks for the requested changes Ali.  Sorry there were so many errors.  In addition I added some punctuation to paragraph starting at line 111 to hopefully make it a bit more clear.

MartinKoob commented:
One additional change I made was to put commas around the 'and' and 'then' in the following sentence to hopefully make it more clear.

> 

 If the <if> <control structure> contains one or more <else if>
<keyword|keywords> which have a <condition|conditional expression>**,
and,** one of those <condition|conditional expressions> evaluates to be
true**, then,** the respective <elseIfstatement|else if statement> or
<elseIfstatementlist|else if statement list> following that
<else if> <keyword> is executed.

MartinKoob commented:
Hmm...  How do you quote from the file in the text of the comment.  I tried copying the text and clicking the 'Quote' icon in the tool bar for the Comment text box.

Martin

livecodeali commented:
I think if something has `<` in it you're best using backticks ``

livecodeali commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -4,9 +4,30 @@ Type: control structure
+<keyword> is executed.
+
+The <if> <control structure> may contain one or more <else if> <keyword|keywords> which have a
+<conditional expression|condition>. If the <conditional expression|condition> one of these evaluates to be
```
There's a couple more `<conditional expression|condition>` that need switching round

livecodeali commented on docs/dictionary/control_st/if.lcdoc:
```diff
@@ -4,9 +4,30 @@ Type: control structure
+
+The <if> <control structure> may contain one or more <else if> <keyword|keywords> which have a
+<conditional expression|condition>. If the <conditional expression|condition> one of these evaluates to be
+true then the <statment|statement> or <statementlist|statement list> following that <else if|else if>
```
this `<else if|else if>` can just be `<else if>`

livecodeali commented:
@MartinKoob I think the commas before the and and then are both fine but there shouldn't be commas afterwards

MartinKoob committed:
Further fixes to MD formatting errors to if.lcdoc

MartinKoob committed:
update if.lcdoc removed extra commas

Removed commas in after 'and' and 'then'

livecodeali commented:
@livecode-vulcan review ok 6ba8f8e

livecode-vulcan commented:
💙 review by @livecodeali ok 6ba8f8e4e5aca92fd88aaa5d70bf044af735277d

livecode-vulcan commented:
😎 test success 6ba8f8e4e5aca92fd88aaa5d70bf044af735277d

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/716)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/732)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/706)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/649)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/652)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/710)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/730)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/715)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/732)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/714)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/725)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/747)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/711)
